cli/src/pcluster/models/imagebuilder.py

```diff
@@ -237,6 +238,13 @@ def config_url(self):
                     raise ImageBuilderActionError(f"Unable to get image {self.image_id} config url.")
         return self.__config_url
 
+    @property
+    def presigned_config_url(self):
+        """Return configuration file as a presigned URL."""
+        if not self.__presigned_config_url:
+            self.__presigned_config_url = create_s3_presigned_url(self.config_url)
+        return self.__presigned_config_url
```

Comment:
have you considered implementing this similarly to cluster without parsing the s3 url?
```
    @property
    def config_presigned_url(self) -> str:
        """Return a pre-signed Url to download the config from the S3 bucket."""
        return self.bucket.get_config_presigned_url(
            config_name=PCLUSTER_S3_ARTIFACTS_DICT.get("source_config_name"),
            version_id=self.stack.original_config_version,
        )
```

Reply:
The stack might not exist if the image has been built and the stack is deleted.

Reply:
(or am I mis-understanding something about the imagebuilder flow)